### PR TITLE
added base structure of common-dao module , added base entities , other...

### PR DESCRIPTION
**Pull request by AlexKazmirchuk**

### CODE REVIEW

File: softgroup/basis/common-dao/common-dao-api/src/main/java/com/softgroup/common/dao/api/entities/ContactEntity.java [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+
+@Entity
+@Table(name = "contacts")
+public class ContactEntity implements Serializable {
+
+    private static final long serialVersionUID = 3180111570947462349L;
+
+    @Id
+    @Column(name = "id")
+    private String id;
```
Comment (Dominys) [inline softgroup/basis/common-dao/common-dao-api/src/main/java/com/softgroup/common/dao/api/entities/ContactEntity.java]: @AlexKazmirchuk Make base class that will hold id

File: softgroup/basis/common-dao/common-dao-api/src/main/java/com/softgroup/common/dao/api/entities/ProfileEntity.java [hunk 2]
```diff
@@ -0,0 +1,133 @@
...
+	@Column(name = "name")
+    private String name;
+
+	@Column(name = "status")
+    private String status;
+
+	@Column(name = "avatar_uri")
+    private String avatarUri;
+
+    @OneToMany(fetch = FetchType.EAGER, mappedBy = "profile")
```
Comment (Dominys) [inline softgroup/basis/common-dao/common-dao-api/src/main/java/com/softgroup/common/dao/api/entities/ProfileEntity.java]: @AlexKazmirchuk You will have many settings ?

File: softgroup/basis/common-dao/common-dao-impl/src/main/java/com/softgroup/common/dao/impl/repositories/ContactRepository.java [hunk 3]
```diff
@@ -0,0 +1,12 @@
...
+
+import com.softgroup.common.dao.api.entities.ContactEntity;
+import org.springframework.data.repository.PagingAndSortingRepository;
+
+/**
+ * @author AlexKazmirchuk
+ * @since 06.03.17.
+ */
+
+public interface ContactRepository extends PagingAndSortingRepository<ContactEntity,String> {
```
Comment (Dominys) [inline softgroup/basis/common-dao/common-dao-impl/src/main/java/com/softgroup/common/dao/impl/repositories/ContactRepository.java]: @AlexKazmirchuk Make base generic repository. All our entities will use String key always.

File: softgroup/basis/common-dao/common-dao-impl/src/main/java/com/softgroup/common/dao/impl/repositories/ProfileRepository.java [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+public interface ProfileRepository extends PagingAndSortingRepository<ProfileEntity, String> {
+
+    List<ProfileEntity> findByName(String s);
+
+    List<ProfileEntity> findByNameAndPhoneNumber(String s, String number);
+
+
+
+    @Query("select p from ProfileEntity p where p.name = :name")
+    List<ProfileEntity> findByNameQuery(@Param("name") String s);
```
Comment (Dominys) [inline softgroup/basis/common-dao/common-dao-impl/src/main/java/com/softgroup/common/dao/impl/repositories/ProfileRepository.java]: @AlexKazmirchuk Copy-paste

File: softgroup/basis/common-dao/common-dao-api/src/main/java/com/softgroup/common/dao/api/entities/ContactEntity.java (hunk 1 above)
Comment (AlexKazmirchuk) [reply softgroup/basis/common-dao/common-dao-api/src/main/java/com/softgroup/common/dao/api/entities/ContactEntity.java]: @Dominys added base abstract class for holding id

File: softgroup/basis/common-dao/common-dao-api/src/main/java/com/softgroup/common/dao/api/entities/ProfileEntity.java (hunk 2 above)
Comment (AlexKazmirchuk) [reply softgroup/basis/common-dao/common-dao-api/src/main/java/com/softgroup/common/dao/api/entities/ProfileEntity.java]: @Dominys  fixed, i will have  oneToOne relation

File: softgroup/basis/common-dao/common-dao-impl/src/main/java/com/softgroup/common/dao/impl/repositories/ContactRepository.java (hunk 3 above)
Comment (AlexKazmirchuk) [reply softgroup/basis/common-dao/common-dao-impl/src/main/java/com/softgroup/common/dao/impl/repositories/ContactRepository.java]: @Dominys done!

File: softgroup/basis/common-dao/common-dao-impl/src/main/java/com/softgroup/common/dao/impl/repositories/ProfileRepository.java (hunk 4 above)
Comment (AlexKazmirchuk) [reply softgroup/basis/common-dao/common-dao-impl/src/main/java/com/softgroup/common/dao/impl/repositories/ProfileRepository.java]: @Dominys  yes(.. deleted now.

File: softgroup/basis/common-dao/common-dao-api/src/main/java/com/softgroup/common/dao/api/entities/BaseEntity.java [hunk 5]
```diff
@@ -0,0 +1,28 @@
...
+
+import javax.persistence.*;
+import java.io.Serializable;
+
+/**
+ * @author AlexKazmirchuk
+ * @since 09.03.17.
+ */
+
+@Entity
```
Comment (Dominys) [inline softgroup/basis/common-dao/common-dao-api/src/main/java/com/softgroup/common/dao/api/entities/BaseEntity.java]: @AlexKazmirchuk This in not Entity

Comment (AlexKazmirchuk) [inline softgroup/basis/common-dao/common-dao-api/src/main/java/com/softgroup/common/dao/api/entities/BaseEntity.java]: @Dominys but without this annotation hybernate cannot create any table for descendants entities
without_annotation

Comment (AlexKazmirchuk) [inline softgroup/basis/common-dao/common-dao-api/src/main/java/com/softgroup/common/dao/api/entities/BaseEntity.java]: with this annotation all works

Comment (AlexKazmirchuk) [reply softgroup/basis/common-dao/common-dao-api/src/main/java/com/softgroup/common/dao/api/entities/BaseEntity.java]: @Dominys fixed

**Review by Dominys — APPROVED** [review]

**Pull request merged**